coev/profiles.go

```diff
@@ -0,0 +1,193 @@
+	MapExtensions extensions.Map
+}
+
+// CoevMapExtensionPoints is a list of extension.Point's valid for a comid.Comid.
+var CoevMapExtensionPoints = []extensions.Point{
+	ExtConciseEvidence,
+	ExtEvTriples,
+	ExtEvidenceTriples,
+	ExtEvidenceTriplesFlags,
+}
+
+// AllExtensionPoints is a list of all valid extension.Point's
+var AllExtensionPoints = make(map[extensions.Point]bool) // populated inside init() below
+
+type iextensible interface {
+	RegisterExtensions(exts extensions.Map) error
+}
+
+// GetCoev returns a pointer to a new comid.Comid that had the ProfileManifest's
+// extensions (if any) registered.
+func (o *ProfileManifest) GetConciseEvidence() *ConciseEvidence {
+	ret := NewConciseEvidence()
+	o.registerExtensions(ret, CoevMapExtensionPoints)
+	return ret
+}
+
+// GetTaggedConciseEvidence returns a pointer to a new aggedConciseEvidence that had the
+// ProfileManifest's extensions (if any) registered.
+func (o *ProfileManifest) GetTaggedConciseEvidence() *TaggedConciseEvidence {
+	r := o.GetConciseEvidence()
+	ret, err := NewTaggedConciseEvidence(r)
+	if err != nil {
+		panic(err)
+	}
+	return ret
+}
+
+var profilesRegister = make(map[string]ProfileManifest)
+
+func (o *ProfileManifest) registerExtensions(e iextensible, points []extensions.Point) {
+	exts := extensions.NewMap()
+	for _, p := range points {
+		if v, ok := o.MapExtensions[p]; ok {
+			exts[p] = v
+		}
+	}
+
+	if err := e.RegisterExtensions(exts); err != nil {
+		// exts is a subset of o.MapExtensions which have been
+		// validated when the profile was registered, so we should never
+		// get here.
+		panic(err)
+	}
+}
+
+// RegisterProfile registers a set of extensions with the specified profile. If
+// the profile has already been registered, or if the extensions are invalid,
+// an error is returned.
+func RegisterProfile(id *eat.Profile, exts extensions.Map) error {
+	strID, err := id.Get()
+	if err != nil {
+		return err
+	}
+
+	if _, ok := profilesRegister[strID]; ok {
+		return fmt.Errorf("profile with id %q already registered", strID)
+	}
+
+	for p, v := range exts {
+		if _, ok := AllExtensionPoints[p]; !ok {
+			return fmt.Errorf("%w: %q", extensions.ErrUnexpectedPoint, p)
+		}
+
+		if reflect.TypeOf(v).Kind() != reflect.Pointer {
+			return fmt.Errorf("attempting to register a non-pointer IMapValue for %q", p)
+		}
+	}
+
+	profilesRegister[strID] = ProfileManifest{ID: id, MapExtensions: exts}
+
+	return nil
+}
+
+// GetProfileManifest returns the ProfileManifest associated with the specified ID, or an empty
+// profileManifest if no ProfileManifest has been registered for the ID. The second return
+// value indicates whether a profileManifest for the ID has been found.
+func GetProfileManifest(id *eat.Profile) (ProfileManifest, bool) {
+	if id == nil {
+		return ProfileManifest{}, false
+	}
+
+	strID, err := id.Get()
+	if err != nil {
+		return ProfileManifest{}, false
+	}
+
+	prof, ok := profilesRegister[strID]
+	return prof, ok
+}
+
+func init() {
```

Comment:
I’d move this to the bottom of the file.  It is where someone would normally look for an `init` function.